### ISSUE TITLE
"odo service list" lists a service that's not present on the cluster

### ISSUE THREAD

**Issue by dharmit** [openshift/odo#1988]
[kind/bug]

## What versions of software are you using?
- Operating System: Fedora (but I doubt all)
- Output of `odo version`: `master`

## How did you run odo exactly?
- I created a service with `odo service create`
- Before the service was provisioned, I did `odo service delete` for the service created in first step

## Actual behavior
- `odo service list` keeps showing the service to be in `Deprovisioning` state.
- `oc get service -n myproject` shows no resources.

## Expected behavior
- `odo service list` should not show the service either.

## Any logs, error output, etc?
```sh
$ oc get services -n myproject
No resources found.

$ odo service list
NAME                  TYPE                  STATUS
dh-prometheus-apb     dh-prometheus-apb     Deprovisioning
```

**Comment by girishramnani** [comment]
does the service disappear after it has completed the `Deprovisioning` state? If so then isn't `odo service list` giving better information then`oc` about the service state? @dharmit

**Comment by dharmit** [comment]
> does the service disappear after it has completed the Deprovisioning state?

It keeps showing the service to be in `Deprovisioning` state forever. As per `odo` output, service doesn't get out of that stage at all.

> If so then isn't odo service list giving better information thenoc about the service state?

Doesn't `odo` use `oc` apis under the hood for operations where it has to get something done on OpenShift cluster? If yes, how could `odo` be a better indicator of the state of things on the cluster than `oc` is? If no, then you're likely right but as I said above, service doesn't get out of `Deprovisioning` state as far as `odo` is concerned while `oc` shows that no resources of the service exist anymore.

**Comment by dharmit** [comment]
Also, I opened this and #1989 at almost the same time. #2008 opened by you has addressed #1989. I guess we should check if this issue is relevant anymore.

**Comment by dharmit** [comment]
```sh
$ odo service create dh-postgresql-apb psql --plan dev -p postgresql_database=mydb -p postgresql_password=password -p postgresql_user=luke -p postgresql_version=10
 ✓  Creating service [58ms]
 ✓  Service 'psql' was created
Progress of the provisioning will not be reported and might take a long time.
You can see the current status by executing 'odo service list'

$ odo service list --app app
NAME     TYPE                  PLAN     STATUS
psql     dh-postgresql-apb     dev      Provisioning

$ odo service delete -f psql
Service psql from application  has been deleted

$ odo service list --app app
NAME     TYPE                  PLAN     STATUS
psql     dh-postgresql-apb     dev      Provisioning

$ odo service list --app app
NAME     TYPE                  PLAN     STATUS
psql     dh-postgresql-apb     dev      Provisioning

$ odo service list --app app
 ✗  There are no services deployed for this application
```

I think it's OK to close this issue.
